src/vs/workbench/contrib/terminal/common/scripts/shellIntegration-rc.zsh

```diff
@@ -32,6 +32,17 @@ if [[ "$VSCODE_INJECTION" == "1" ]]; then
 	fi
 fi
 
+zsh_version=${ZSH_VERSION}
+use_associative_array=0
+vsc_env_keys=()
+vsc_env_values=()
+
+# Associative array are only available in zsh 4.3 or later
+if (( zsh_version >= 4.3 )); then
```

Comment:
`zsh_version` would be better to inline inside the if so we don't have the variable sticking around